backend/src/main/java/io/castles/core/service/LobbyService.java

```diff
@@ -21,19 +25,40 @@ public LobbyService(Server server, SseEmitterService emitterService) {
     }
 
     public void joinLobby(UUID id, Player player) throws IOException {
+
         var gameLobby = server.gameLobbyById(id);
         gameLobby.addPlayer(player); // TODO: exception handling
 
         var playerId = player.getId();
         emitterService.createPlayerEmitterForLobby(id, playerId);
 
-        updateLobbyState(gameLobby);
+        updateLobbyState(gameLobby.getId());
+    }
+
+    public SseEmitter reconnectLobby(UUID id, UUID playerId) {
+        var gameLobby = server.gameLobbyById(id);
+        if (!gameLobby.isPlayerInLobby(playerId)) {
+            throw new NoSuchElementException(String.format("No player with id %s found in lobby %s", playerId, id));
+        }
+        this.emitterService.createPlayerEmitterForLobby(id, playerId);
+        return this.emitterService.getLobbyEmitterForPlayer(id, playerId);
     }
 
-    public void updateLobbyState(GameLobby gameLobby) throws IOException {
+    public void updateLobbyState(UUID id) {
+        var gameLobby = server.gameLobbyById(id);
         var lobbySseEmitters = this.emitterService.getAllLobbyEmitters(gameLobby.getId());
         for (var playerSseEmitter : lobbySseEmitters) {
-            playerSseEmitter.send(LobbyStateDTO.from(gameLobby));
+            try {
+                playerSseEmitter.send(LobbyStateDTO.from(gameLobby));
+            } catch (IOException e) {
+                lobbySseEmitters.remove(playerSseEmitter);
+            }
```

Comment:
Every IOException is cached and the throwing playerSseEmitter is removed from the lobbies emitters.
This connection is restored for the subscribing player in the Controller.